### PR TITLE
feat!: remove store ID from API config, allow override per-request

### PR DESCRIPTION
**Pull request by jimmyjames**
Removes `StoreId` from `api_open_fga.go` configuration, and enables per-request override of the `StoreId`. 

## Description

⚠️ Breaking change ⚠️

Removes the `StoreId` from the API client, and instead accept the `StoreId` as a parameter. As discussed in #118, this allows for the removal of the custom patch during SDK generation. 

**If using api_open_fga.go directly, you will now need to pass the StoreId parameter.**

This change also enables per-request override of the `StoreId`, via the associated request's options. 

## References
* https://github.com/openfga/sdk-generator/issues/118
* https://github.com/openfga/sdk-generator/pull/359
* Generated from https://github.com/openfga/sdk-generator/pull/364

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [x] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -773,8 +773,6 @@ List the users who have a certain relation to a particular type.
 options := ClientListRelationsOptions{
     // You can rely on the model id set in the configuration or override it for this specific request
     AuthorizationModelId: openfga.PtrString("01GAHCE4YVKPQEKZQHT2R89MQV"),
-    // Max number of requests to issue in parallel, defaults to 10
```
Comment (ewanharris) [inline README.md]: It looks like the change that added this hadn't made it across to sdk-generator. Created openfga/sdk-generator/pull/365

**Review by ewanharris — COMMENTED** [review]
Nice work! Changes in general look good, just the one callout where we're dropping #93 as it wasn't backported. Maybe we can merge openfga/sdk-generator/pull/365 and regenerate these changes?

File: README.md (hunk 1 above)
Comment (jimmyjames) [reply README.md]: Great catch! I'll pull those changes into my generator change and update this PR after running the gen again.

File: client/client.go [hunk 2]
```diff
@@ -461,6 +464,34 @@ func (client *OpenFgaClient) getAuthorizationModelId(authorizationModelId *strin
...
+		return FgaInvalidError{param: "StoreId", description: "ULID"}
+	}
+	client.config.StoreId = storeId
+	return nil
+}
+
+func (client *OpenFgaClient) GetStoreId() (string, error) {
+	storeId := client.config.StoreId
+	if storeId != "" && !internalutils.IsWellFormedUlidString(storeId) {
+		return "", FgaInvalidError{param: "StoreId", description: "ULID"}
```
Comment (rhamzeh) [inline client/client.go]: I know this is from before, but can we make this more descriptive? Maybe in a follow up PR?

Something like "expected to match ULID format" or something

Comment (jimmyjames) [reply client/client.go]: I updated the description for all cases where it was just "ULID" to "expected ULID format".

File: README.md (hunk 1 above)
Comment (jimmyjames) [reply README.md]: I think we're good now, but please check ;). I made https://github.com/openfga/sdk-generator/pull/367 to patch back some of the README changes that were missed.

**Review by ewanharris — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (ewanharris) [reply README.md]: Oops, thanks for catching that!

**Review by rhamzeh — APPROVED** [review]

**Pull request merged**